### PR TITLE
Fix AMD modules which return falsy values

### LINKED ISSUE #2129: AMD modules cannot return falsy values
Hi systemjs team,

If I create AMD modules such as these:
```
define([], function () {
  return false;
});

define([], function () {
  return '';
});
```
when they're imported, the default export is an empty object instead of `false` or `''`.

The values are thrown out here:
https://github.com/systemjs/systemjs/blob/6caaee592172954bbf65bd8490212286f6e38754/src/extras/amd.js#L58

I'll submit a PR with a fix for your approval: #2130

### PR DESCRIPTION
**Pull request by lpomerleau**
- fixes #2129 
- AMD modules can successfully return falsy values such as `''` or `false`

### CODE REVIEW

File: dist/extras/amd.js [hunk 1]
```diff
@@ -55,7 +55,9 @@
       return {
         setters: setters,
         execute: function () {
-          module.exports = amdExec.apply(exports, depModules) || module.exports;
+          const amdResult = amdExec.apply(exports, depModules);
+          if (amdResult !== null && amdResult !== undefined)
```
Comment (dmail) [inline dist/extras/amd.js]: Why testing both null and undefined? Undefined is not enough ?
I'm just passing by and wondering don't feel forced to answer :)

Comment (lpomerleau) [reply dist/extras/amd.js]: Agreed, I've changed it to just check for `undefined`, unless someone tells me that `null` is a possibility (eg: deletion edge case?).

**Comment by joeldenning** [comment]
Released in https://github.com/systemjs/systemjs/releases/tag/6.2.5

**Comment by lpomerleau** [comment]
Thanks for the quick response!

**Pull request merged**